### ISSUE TITLE
pu=strftime "en août" breaks fileencoding

### ISSUE THREAD

**Issue by harriott** [neovim/neovim#30013]
### Problem

In August, on a French system, in a `utf-8` file, `:pu=strftime('%B')` puts `ao<fb>t` where that `<fb>` is some kind of holding character. Saving and reloading the file then switches it to `fileencoding=latin1`, and un-undoably replaces existing `utf-8` glyphs with multi-character `latin1` encodings.

### Steps to reproduce

nvim --clean empty_utf8_file
:se fenc=utf8
:pu=strftime('%B')
:w
:e
:se fenc

### Expected behavior

In August in France, `:pu=strftime('%B')` should `:put` "août" with no consequent switching of `fileencoding` to `latin1`.

### Neovim version (nvim -v)

NVIM v0.10.1

### Vim (not Nvim) behaves the same?

no, vim 9.1

### Operating system/version

Windows 10 Pro

### Terminal name/version

Windows Terminal v1.20.11781.0

### $TERM environment variable

not applicable

### Installation

winget add Neovim.Neovim

**Comment by clason** [comment]
what is your _exact_ locale?

**Comment by neovim-discourse** [comment]
This issue has been mentioned on **Neovim Discourse**. There might be relevant details there:

https://neovim.discourse.group/t/pu-strftime-en-aout-breaks-fileencoding/5043/1

**Comment by harriott** [comment]
:language
    Current language: "LC_COLLATE=fr_FR;LC_CTYPE=fr_FR;LC_MONETARY=fr_FR;LC_NUMERIC=C;LC_TIME=fr_FR"

    Windows 10 Pro > Paramètres > Heure et langue >
        Region > France
        Langue > Français (France)

**Comment by clason** [comment]
Yeah, those are invalid. You _must_ add the encoding, so `fr_FR.UTF-8`.
